src/functions/getActivity.ts

```diff
@@ -1,18 +1,29 @@
 import { GetActivityService } from './../services/GetActivitiesService';
-import { Context, Handler } from 'aws-lambda';
+import { Context } from 'aws-lambda';
 import { HTTPResponse } from '../utils/HTTPResponse';
 import { DynamoDBService } from '../services/DynamoDBService';
+import { HTTPRESPONSE } from '../assets/enums';
+
+export async function getActivity(event: any, context?: Context): Promise<any> {
+  console.log('getActivites');
+  if (!(event && event.queryStringParameters)) {
+    return new HTTPResponse(400, HTTPRESPONSE.BAD_REQUEST);
+  }
 
-const getActivity: Handler = async (event: any, context?: Context): Promise<any> => {
   const activityService = new GetActivityService(new DynamoDBService());
-  return activityService
-    .getActivities(event)
-    .then((data: any) => {
-      return new HTTPResponse(200, data);
-    })
-    .catch((error: HTTPResponse) => {
-      return error;
-    });
+  const { fromStartTime, toStartTime, activityType, testStationPNumber, testerStaffId } =
+    event.queryStringParameters && event.queryStringParameters;
```

Comment:
Why is event.queryStringparameters in there twice?

Reply:
It is used to short-circuit and assign `queryStringParameters`. We can remove this though as we are checking `queryStringParameters` existence on line 9.